### PR TITLE
Rewards check includes

### PR DESCRIPTION
**Pull request by bridiver**
Resolves https://github.com/brave/brave-browser/issues/10618

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by zenparsing — APPROVED** [review]
This is great - thanks!

File: browser/brave_rewards/rewards_tab_helper.cc [hunk 1]
```diff
@@ -21,14 +21,36 @@
...
 #include "chrome/browser/ui/browser_finder.h"
 #include "chrome/browser/ui/browser_list.h"
+#include "chrome/browser/ui/browser_list_observer.h"
 #include "chrome/browser/ui/tabs/tab_strip_model.h"
 #endif
 
 namespace brave_rewards {
 
+#if !BUILDFLAG(IS_ANDROID)
+class BraveBrowserListObserver : public BrowserListObserver {
```
[the remark this answers is not in the document]
Comment (bridiver) [reply browser/brave_rewards/rewards_tab_helper.cc:31]: I wasn't really sure what to call it, but I'm open to suggestions

File: browser/brave_rewards/BUILD.gn [hunk 2]
```diff
@@ -3,8 +3,70 @@
...
 # You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import("//brave/components/greaselion/browser/buildflags/buildflags.gni")
+import("//extensions/buildflags/buildflags.gni")
 import("//testing/test.gni")
 
+source_set("brave_rewards") {
+  sources = [
+    "//brave/browser/brave_rewards/rewards_service_factory.h",
+    "//brave/browser/brave_rewards/rewards_tab_helper.h",
```
Comment (goodov) [inline browser/brave_rewards/BUILD.gn:13]: nit: consider fixing these paths to be local in all new targets you introduce in this PR:
```
    "rewards_service_factory.h",
    "rewards_tab_helper.h",
```

File: browser/brave_rewards/rewards_tab_helper.cc [hunk 3]
```diff
@@ -21,14 +21,36 @@
...
+  ~BraveBrowserListObserver() override {}
+  void OnBrowserSetLastActive(Browser* browser) override {
+    tab_helper_->OnBrowserSetLastActive(browser);
+  }
+  void OnBrowserNoLongerActive(Browser* browser) override {
+    tab_helper_->OnBrowserNoLongerActive(browser);
+  }
+
+ private:
+  raw_ptr<RewardsTabHelper> tab_helper_;  // Not owned.
```
Comment (goodov) [inline browser/brave_rewards/rewards_tab_helper.cc:44]: `const` all constructor-initialized pointers to trigger compile errors if initialization is missing.

File: browser/ui/BUILD.gn [hunk 4]
```diff
@@ -1374,6 +1375,23 @@ source_set("brave_tab_features") {
   deps = [ "//base" ]
 }
 
+source_set("brave_rewards_source") {
+  sources = [
+    "webui/brave_rewards_source.cc",
+    "webui/brave_rewards_source.h",
```
Comment (goodov) [inline browser/ui/BUILD.gn:1381]: there's `ui/webui/brave_rewards` dir. Would be nice to move this into it.

Comment (bridiver) [reply browser/ui/BUILD.gn:1381]: trying to avoid any changes that aren't necessary here because it already touches a lot of stuff

**Review by darkdh — APPROVED** [review]

File: components/brave_rewards/browser/BUILD.gn [hunk 5]
```diff
@@ -41,40 +38,44 @@ static_library("browser") {
...
+    "//brave/components/brave_rewards/core",
+    "//brave/components/brave_wallet/browser",
+    "//brave/components/brave_wallet/common",
     "//brave/components/greaselion/browser/buildflags",
+    "//brave/components/ntp_background_images/common",
+    "//brave/components/p3a_utils",
     "//brave/components/resources",
     "//brave/components/services/bat_rewards/public/interfaces",
+    "//brave/components/time_period_storage",
+    "//chrome/browser/bitmap_fetcher",  # bad
```
Comment (bridiver) [inline components/brave_rewards/browser/BUILD.gn:52]: oops, I totally missed this before I merged it. Need to fix this one

**Pull request merged**